### ISSUE TITLE
Scores of different RAG methods in the data of Table 3 (Llama3-8B)

### ISSUE THREAD

**Issue by BlueCestbon** [fate-ubw/RAGLAB#6]
I noticed the scores in the data of Table 3 (Llama3-8B), and found that the scores of Direct and NaiveRAG are higher than the remaining 5 methods. However, in the corresponding method papers, their scores are much higher than this. Even if the test data set is 500 random data, the data distribution should be consistent. Why is there such a big difference? Is it because of the specific fine-tuning of the Llama3-8B model in this framework? In addition, why didn't you use the model of the Llama3-8B-Instruct series? Is it for fairness?
![image](https://github.com/user-attachments/assets/da4b39ab-9a1a-4b53-a624-a4a898cd3e59)

**Comment by fate-ubw** [comment]
Thank you very much for your questions. I'll answer them one by one.

Q1: The scores of Direct and NaiveRAG are higher than the remaining 5 methods. Why is there such a big difference?

A: I'm glad you pointed out this valuable question. I noticed this issue immediately when we got the experimental results. Here's my response, broken down into points:

1. Using llama3-8B-baseline as the generator, we were surprised to find that Direct and NaiveRAG scored higher than the remaining 5 methods. We tried experimenting with other generators like llama2-chat and got similar results. We also used llama3-70B-baseline and chatgpt3.5 as generators and obtained similar experimental results. Therefore, we believe that NaiveRAG itself is a highly robust baseline. Although flashrag didn't completely align with our generator, it also reached a similar conclusion. You can see in the table that naiverag shows strong performance.
    <img width="1104" alt="image" src="https://github.com/user-attachments/assets/0b919fa6-b1ca-4d09-a640-a104846cc1e6">

2. In our experiments, we set the use of top-10 passages as external knowledge each time. It's very likely that the length of the top-10 passages was too long, introducing a lot of noise. This may have made it difficult for llama3-8B-baseline and chatgpt3.5 to accurately capture key information, ultimately resulting in Direct and NaiveRAG scoring higher than the remaining 5 methods.
3. Regarding the impact of sampling 500 instances on the results, we found in multiple experiments that different sampling methods significantly affect the results. We previously experimented with evaluating 1000 data points and found considerable differences compared to measuring 500, especially for the PopQA dataset. So I believe 500 is far from enough. However, many papers only use 500 data points when evaluating their proposed algorithms, and the sampling methods differ. REACT randomly samples 500 data points each time, while some papers don't mention how they sample data. We used sequential sampling of the first 500 data points, which makes it easier to reproduce the same results. If randomly sampled, the distribution difference between evaluation data would be quite large. Due to time and computational power limitations, we ultimately chose 500 data points, but I think measuring 1000 to 2000 data points would likely yield more accurate results.
4. Another crucial issue is that I believe there are some problems with the evaluation metrics. The acc (accuracy) metric requires the output to strictly match the label, which means many correct answers are considered wrong due to the calculation method of acc. For example, differences in capitalization can cause an answer to be marked as incorrect. Therefore, I think using gpt4o to judge whether an answer is correct would be a more accurate method, although the downside is that it's more expensive.
5. To increase the credibility of our experiments, we have open-sourced all log files and evaluation results. Please refer to: https://huggingface.co/datasets/RAGLAB/raglab-eval_results

Q2: Is it because of the specific fine-tuning of the Llama3-8B model in this framework?

A: Initially, I thought it was related to the base model. However, since we also used chatgpt3.5 as a generator and still concluded that the scores of Direct and NaiveRAG are higher than the remaining 5 methods, the most likely conclusion is that the remaining 5 methods are not as effective as we originally thought. Furthermore, when reproducing many algorithms, I often find that there are some tricks in the officially open-sourced code implementations, which may also be a reason for the different results. 

Q3: In addition, why didn't you use the model of the Llama3-8B-Instruct series? Is it for fairness?

A: In our first experiment, we also used Llama3-8B-Instruct as the base model, but we found that the results obtained from selfrag-Llama3-8B-Instruct fine-tuning were not good. The performance of selfrag-Llama3-8B-Instruct was lower than that of selfrag-Llama3-8B. We fine-tuned many models to verify this conclusion. Additionally, the selfrag paper mentions that selfrag_llama_7b uses the meta-llama/Llama-2-7b-hf model for fine-tuning. To maintain consistency with the paper, we ultimately decided to use Llama3-8B as the base model. Therefore, using Llama3-8B as the base model was mainly because we strictly reproduced the paper's settings. However, we still encountered the first two problems: selfrag did not show very good performance on the 8B model, but selfrag-70B model showed excellent performance. If you have also trained selfrag, we would very much welcome you to share your experimental results.

Feel free to ask more questions. I'm happy to answer any queries you may have~

**Comment by BlueCestbon** [comment]
Thank you very much for your detailed and serious answer, because I have reproduced several of the methods in Llama2-7b-chat and Llama3-8B-Instruct before, and was surprised to find that there was not much difference between them and Naive's method. Now after reading your reply, I think there may indeed be some differences in data sampling.
Finally, thank you again for your analysis and reply.
